src/os/bluestore/BlueStore.cc

```diff
@@ -5588,7 +5587,7 @@ void BlueStore::_do_write_small(
     }
     int64_t blob = ep->second.blob;
     b = c->get_blob(o, ep->second.blob);
-    if (!b->blob.is_mutable()) {
+    if (!b->blob.is_mutable() || b->blob.is_compressed()) {
```

Comment:
this should be redundant.. anything that is compressed will necessarily not have mutable flag set.